Email validation blocks upper case letters

LukaszRacon opened the pull request:
Fix makes email validation regex case insensitive.

lukaszlenart commented on core/src/main/resources/template/xhtml/form-close-validate.ftl:
```diff
@@ -114,7 +114,7 @@ END SNIPPET: supported-validators
             }
             <#elseif validator.validatorType = "email">
-            if (continueValidation && fieldValue != null && fieldValue.length > 0 && fieldValue.match("${validator.regex?js_string}")==null) {
+            if (continueValidation && fieldValue != null && fieldValue.length > 0 && fieldValue.match(new RegExp("${validator.regex?js_string}", "i"))==null) {
```
What about using simplified form and match the same case few lines below?
```
"/${validator.regex?js_string}/i"
```

LukaszRacon commented:
I updated PR with a shorter literal regexp notation.

`"/regexString/i"` will not work since it would be converted by str.match function to RegExp object
`new RegExp("/regexString/i")` while we need RegExp with a case-insensitive flag: `new RegExp("regexString", "i")`

So in this case we could use:
```
fieldValue.match(new RegExp("${validator.regex?js_string}", "i"))==null
```
or shorter version:
```
fieldValue.match(/${validator.regex}/i)==null
```
I also fixed url validation that used incorrect RegExp.
